diffprivlib/models/naive_bayes.py

```diff
@@ -74,33 +81,99 @@ class GaussianNB(sk_nb.GaussianNB):
+            self.bounds = (np.min(X, axis=0), np.max(X, axis=0))
+
+        self.bounds = check_bounds(self.bounds, shape=X.shape[1])
+        X = clip_to_bounds(X, self.bounds)
+
+        self.epsilon_ = self.var_smoothing
+
+        if _refit:
+            self.classes_ = None
+
+        if _check_partial_fit_first_call(self, classes):
+            n_features = X.shape[1]
+            n_classes = len(self.classes_)
+            self.theta_ = np.zeros((n_classes, n_features))
+            self.sigma_ = np.zeros((n_classes, n_features))
+
+            self.class_count_ = np.zeros(n_classes, dtype=np.float64)
+
+            if self.priors is not None:
+                priors = np.asarray(self.priors)
+
+                if len(priors) != n_classes:
+                    raise ValueError("Number of priors must match number of classes.")
+                if not np.isclose(priors.sum(), 1.0):
+                    raise ValueError("The sum of the priors should be 1.")
+                if (priors < 0).any():
+                    raise ValueError("Priors must be non-negative.")
+                self.class_prior_ = priors
+            else:
+                # Initialize the priors to zeros for each class
+                self.class_prior_ = np.zeros(len(self.classes_), dtype=np.float64)
+        else:
+            if X.shape[1] != self.theta_.shape[1]:
+                msg = "Number of features %d does not match previous data %d."
+                raise ValueError(msg % (X.shape[1], self.theta_.shape[1]))
+            # Put epsilon back in each time
+            self.sigma_[:, :] -= self.epsilon_
+
+        classes = self.classes_
+
+        unique_y = np.unique(y)
+        unique_y_in_classes = np.in1d(unique_y, classes)
+
+        if not np.all(unique_y_in_classes):
+            raise ValueError("The target label(s) %s in y do not exist in the initial classes %s" %
+                             (unique_y[~unique_y_in_classes], classes))
+
+        noisy_class_counts = self._noisy_class_counts(y)
 
-        self.bounds = _check_bounds(self.bounds, X.shape[1])
+        for _i, y_i in enumerate(unique_y):
+            i = classes.searchsorted(y_i)
+            X_i = X[y == y_i, :]
 
-        super()._partial_fit(X, y, classes, _refit, sample_weight=None)
+            n_i = noisy_class_counts[_i]
+
+            new_theta, new_sigma = self._update_mean_variance(self.class_count_[i], self.theta_[i, :],
+                                                              self.sigma_[i, :], X_i, n_noisy=n_i)
+
+            self.theta_[i, :] = new_theta
+            self.sigma_[i, :] = new_sigma
+            self.class_count_[i] += n_i
+
+        self.sigma_[:, :] += self.epsilon_
+
+        # Update if only no priors is provided
+        if self.priors is None:
+            # Empirical prior, with sample_weight taken into account
+            self.class_prior_ = self.class_count_ / self.class_count_.sum()
+
+        self.accountant.spend(self.epsilon, 0)
```

Comment:
I totally agree that the current method of calling `spend()` is not ideal. In an ideal world, we could track privacy spend through the individual mechanisms, but we would run into problems in some instances because a lot of those operations are using parallel composition, which is not (yet) supported by the BudgetAccountant. I'll open an issue about this for future releases.